lib/network/modules/components/edges/util/edge-base.ts

```diff
@@ -106,6 +107,21 @@ export abstract class EdgeBase<Via = undefined> implements EdgeType {
    */
   public setOptions(options: EdgeOptions): void {
     this.options = options;
+
+    if (Object.prototype.hasOwnProperty.call(options, "selfReferenceSize")) {
+      console.log(
+        "The selfReferenceSize property has been deprecated. Please use selfReference property instead. The selfReference can be set like thise selfReference:{size:30, angle:null}"
+      );
+      if (
+        !Object.prototype.hasOwnProperty.call(options, "selfReference") ||
+        !Object.prototype.hasOwnProperty.call(options.selfReference, "size")
+      ) {
+        this.options.selfReference = {
+          size: options.selfReferenceSize
+        };
+      }
+    }
+
```

Comment:
This is redundant. It was already done in `Edge.parseOptions()`.

Reply:
@Thomaash  Are you sure? 
I modified it to be as the following:

if (Object.prototype.hasOwnProperty.call(options, "selfReferenceSize")) {
      if (
        !Object.prototype.hasOwnProperty.call(options, "selfReference") ||
        !Object.prototype.hasOwnProperty.call(options.selfReference, "size")
      ) {
        this.options.selfReference = {
          size: options.selfReferenceSize,
          angle: null
        };
      }
    }


Because if we remove it then unit tests need to be changed from selfReferenceSize to selfReference: {size, angle}, because passed option does not contain selfReference. Unless i missed something.

Reply:
In production this should never get raw options only the options that were processed by `Edge.parseOptions()` (any other behavior is a bug). If the tests test this directly then they should be updated to test the new API not the deprecated one.